contracts/deploy/mainnet/119_pool_booster_curve.js

```diff
@@ -10,54 +13,118 @@ module.exports = deploymentWithGovernanceProposal(
     deployerIsProposer: false,
     proposalId: "",
   },
-  async ({ deployWithConfirmation, withConfirmation }) => {
+  async ({ withConfirmation }) => {
     const { deployerAddr } = await getNamedAccounts();
     const sDeployer = await ethers.provider.getSigner(deployerAddr);
-    console.log(`Using deployer account: ${deployerAddr}`);
-    // 1. Deploy proxy
-    const dCurvePoolBoosterProxy = await deployWithConfirmation(
-      "CurvePoolBoosterProxy"
+    console.log(`Deployer address: ${deployerAddr}`);
+
+    const cCurvePoolBooster = await ethers.getContractAt(
+      "CurvePoolBooster",
+      addresses.zero
     );
-    const cCurvePoolBoosterProxy = await ethers.getContract(
-      "CurvePoolBoosterProxy"
+    const cCurvePoolBoosterProxy = await ethers.getContractAt(
+      "CurvePoolBoosterProxy",
+      addresses.zero
     );
 
-    // 2. Deploy implementation
-    const dCurvePoolBoosterImpl = await deployWithConfirmation(
-      "CurvePoolBooster",
-      [
-        42161, // Arbitrum chain id
-        addresses.mainnet.CampaignRemoteManager, // Campaign Remote Manager (VotemarketV2 entry point)
-        addresses.mainnet.OUSDProxy, // Bribe token
-        addresses.mainnet.CurveOUSDUSDTGauge, // Gauge
-      ]
+    console.log("\nStarting deployment using CreateX");
+    const rewardToken = addresses.mainnet.OUSDProxy;
+    const gauge = addresses.mainnet.CurveOUSDUSDTGauge;
+    const targetedChainId = 42161; // arbitrum
+    const fee = 0;
+
+    // Get CreateX contract
+    const cCreateX = await ethers.getContractAt(createxAbi, addresses.createX);
+
+    // Generate encoded salt (deployer address || crossChainProtectionFlag || bytes11(keccak256(rewardToken, gauge)))
+    const addressDeployerBytes20 = ethers.utils.hexlify(
+      ethers.utils.zeroPad(deployerAddr, 20)
     );
-    console.log("dCurvePoolBoosterImpl: ", dCurvePoolBoosterImpl.address);
-    const cCurvePoolBooster = await ethers.getContractAt(
-      "CurvePoolBooster",
-      dCurvePoolBoosterProxy.address
+    const crossChainProtectioFlagBytes1 = ethers.utils.hexlify(
+      ethers.utils.zeroPad(0, 1)
+    );
+    const saltBytes11 =
+      "0x" +
+      ethers.utils
+        .keccak256(
+          ethers.utils.concat([
+            ethers.utils.arrayify(rewardToken),
+            ethers.utils.arrayify(gauge),
+          ])
+        )
+        .slice(2, 24);
+    const encodedSalt = ethers.utils.hexlify(
+      ethers.utils.concat([
+        addressDeployerBytes20,
+        crossChainProtectioFlagBytes1,
+        saltBytes11,
+      ])
     );
+    console.log(`Encoded salt: ${encodedSalt}`);
 
-    // 3. Initialize
-    const initData = cCurvePoolBooster.interface.encodeFunctionData(
-      "initialize(address,uint16,address)",
+    // --- Deploy implementation --- //
+    const cachedInitCodeImpl = ethers.utils.concat([
```

Comment:
Pretty cool that you've made this work with the IMPL contract as well. Also nice to see first hand how the constructor parameters are just encoded and appended to the contract bytecode. 

Just wanted to leave a remark, that we don't really need matching implementation addresses cross chain. And in favor of less complexity we shouldn't probably do it in the future. I am fine though leaving this one is as it is

Reply:
Indeed, matching implementation doesn't bring any value, 100% agree. (it's just even more satisfying).
 
Since I'd already started, I thought I might as well continue with the same method, in order to have an exemple of it done once (if for any reason we could need it in the futur).

I prefer to keep it like this for this implementation, but 100% agree on just using CreateX for Proxy for next time 👍

Reply:
agreed :)